### PR TITLE
Update font tooling to support semi-bold font weight

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Updates fontTools library to the latest version, and enforce Python version at a module level in the Python aspects of the i18n module
* Updates our font tooling to use the new repository https://github.com/notofonts/notofonts.github.io as the old one has been archived
* Updates our font tooling to either download SemiBold weights, or in the case where they are unavailable, use variable versions to generate the missing font weight (or weights).
* Note that this means that some of our fonts will be unhinted, whereas until now all our fonts are hinted.
* Preference is given to downloading the slimline version of the variable fonts (as we do not need variation in font width) but it falls back to the full variable font if not available.
* Adds the ability to exclude specific typefaces that would otherwise be bundled based on our 'weights are available' criteria - does so with the NotoSansTest font which is not needed and is only for testing.
* Excludes fonts that are only used for ancient texts, or have incredibly limited usage.
* Adds some specific exclusions for fonts used during font subsetting - these fonts seemed to cause errors during font subsetting that I could not resolve, and as they are not used in UI text, it seemed simplest to exclude them from the subsetting code.

## Reviewer guidance
In the future we _may_ want to be explicit about which typefaces we want to include in the font subsetting code - although, with the dropping of IE11 support in the future, it's possible we can stop doing the subsetting altogether.

Opening this as a draft for now, to make sure that everything builds, check the size of the built WHL file, and also open it up for QA testing to make sure fonts are still behaving as expected.

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by radinamatic** [comment]
Tested the `DEB` asset in Firefox and Chrome on Ubuntu 20.04, and checked the loading of all 32 locales in Kolibri. No issues noticed neither in the UI font display, nor in the network tab and console regarding the subsets loading.

This newest `DEB` is 81.7 MB compared to 75.3 MB of 0.16.0 release.



@pcenov can you please test the `EXE` and `DMG` assets? Thank you!

**Comment by pcenov** [comment]
Hi @radinamatic no fonts related issues observed while testing with the .deb and .exe builds as well.

**Review by radinamatic — APPROVED** [review]
Manual QA passes, should be good to go after the code review! 🙂

File: packages/kolibri-tools/lib/i18n/fonts.py
Comment (nucleogenesis) [inline packages/kolibri-tools/lib/i18n/fonts.py]: Why do we notify that we've updated fontTools on one type of error but not the other?

**Review by nucleogenesis — APPROVED** [review]
I think I got a good grasp of what was going on in the code and the changes made sense and made for a clean grokkable read.

Just left a non-blocking question

File: packages/kolibri-tools/lib/i18n/fonts.py
Comment (rtibbles) [reply packages/kolibri-tools/lib/i18n/fonts.py]: The assertion error only happens in the case that fontTools was already installed. Unfortunately, because we already imported it, we have to exit Python before we can import the updated version (although, now I think about it, it _might_ be possible to clear the module cache - but generally safer to quit). When the import didn't even succeed, we can reimport again.

**Review by marcellamaki — DISMISSED** [review]
Blocking merge into 0.16.x until we are ready for all branding to be merged

**Pull request merged**
